### PR TITLE
Add app_deployment role

### PR DESCRIPTION
**Pull request by machacekondra**
Let's add roles here, and once collections repo is ready, let's move it there, so we can review the changes and test them.

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

File: roles/app_deployment/tasks/vms.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    wait: "{{ item.wait | default(true) }}"
+    name: "{{ item.name | mandatory }}"
+    namespace: "{{ item.namespace | default('default') }}"
+    memory: "{{ item.memory | default(omit) }}"
+    template: "{{ item.template | default(omit) }}"
+    disks: "{{ item.disks | default(omit) }}"
+    interfaces: "{{ item.interfaces | default(omit) }}"
+    cloud_init_nocloud: "{{ item.cloud_init_nocloud | default(omit) }}"
+    datavolumes: "{{ dvs }}"
+    wait_timeout: 240
```
Comment (pkliczewski) [inline roles/app_deployment/tasks/vms.yml]: Should we parametrize timeout?

File: roles/app_deployment/tasks/vms.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+          storage: "{{ item.image.storage }}"
+  when: item.image is defined
+
+- name: Create virtual machines
+  kubevirt_vm:
+    state: "{{ item.state | default('running') }}"
+    wait: "{{ item.wait | default(true) }}"
+    name: "{{ item.name | mandatory }}"
+    namespace: "{{ item.namespace | default('default') }}"
+    memory: "{{ item.memory | default(omit) }}"
```
Comment (pkliczewski) [inline roles/app_deployment/tasks/vms.yml]: What about cpu cores?

File: roles/app_deployment/tasks/vms.yml (hunk 1 above)
Comment (machacekondra) [reply roles/app_deployment/tasks/vms.yml]: Definitelly.

File: roles/app_deployment/tasks/vms.yml (hunk 2 above)
Comment (machacekondra) [reply roles/app_deployment/tasks/vms.yml]: This is basic layout WIP. I will be adding other stuff, like cloud init, affinity, network etc.

File: tests/playbooks/library/kubevirt_vm.py [hunk 3]
```diff
@@ -0,0 +1,454 @@
+#!/usr/bin/python
```
Comment (machacekondra) [inline tests/playbooks/library/kubevirt_vm.py]: Remove when https://github.com/ansible/ansible/pull/56846 is merged.

File: tests/playbooks/module_utils/kubevirt.py [hunk 4]
```diff
@@ -0,0 +1,404 @@
+# -*- coding: utf-8 -*-
```
Comment (machacekondra) [inline tests/playbooks/module_utils/kubevirt.py]: Remove when https://github.com/ansible/ansible/pull/56846 is merged.

File: roles/app_deployment/tasks/vms.yml (hunk 2 above)
Comment (machacekondra) [reply roles/app_deployment/tasks/vms.yml]: My idea is to merge this and add each feature in seprate PR so you can participate on small code reviews.

File: tests/playbooks/library/kubevirt_preset.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/python
```
Comment (machacekondra) [inline tests/playbooks/library/kubevirt_preset.py]: Remove. when https://github.com/ansible/ansible/pull/56865/files is merged

File: tests/playbooks/library/kubevirt_rs.py [hunk 6]
```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python
```
Comment (machacekondra) [inline tests/playbooks/library/kubevirt_rs.py]: Remove. when https://github.com/ansible/ansible/pull/56865/files is merged

**Comment by machacekondra** [comment]
Ready for review.

**Pull request merged**